Makefile

```diff
@@ -64,20 +64,24 @@ pull: ## update the Docker image used by "make shell"
 	docker pull edxops/edxapp:latest
 
 pre-requirements: ## install Python requirements for running pip-tools
+	pip install -qr requirements/pip.txt
 	pip install -qr requirements/edx/pip-tools.txt
 
 local-requirements:
 # 	edx-platform installs some Python projects from within the edx-platform repo itself.
 	pip install -e .
 
-dev-requirements: local-requirements
-	pip install -qr requirements/edx/development.txt
+dev-requirements: pre-requirements
+	pip-sync requirements/edx/development.txt
+	make local-requirements
 
-base-requirements: local-requirements
-	pip install -qr requirements/edx/base.txt
+base-requirements: pre-requirements
+	pip-sync requirements/edx/base.txt
+	make local-requirements
 
-test-requirements: local-requirements
-	pip install --exists-action='w' -qr requirements/edx/testing.txt
```

Comment:
This is the important line, since this target is what's used in `unit-tests.yml` and is where the stale virtualenv issue was coming from.

Reply:
Incidentally, I'm not totally sure why `--exists-action=w` is needed here, but if it's not included, then some `-e` dependencies fail to be installed properly when this is run from GH actions.

Reply:
Just checking my understanding: the problem was that when we called local_requirements as a prereq, said requirements were getting overwritten by the next line because that line used pip instead of pip-sync?

Reply:
Sort of. Local-requirements was running and installing stuff, and then testing.txt was installing on top of it. There may have been some overwriting going on. But the real problem was that all of this was happening on `edx-platform-runner` instances that were getting _reused_ (I may need to make this clearer in the commit message) and there was nothing removing dependencies from that stale virtualenv when setting up for a new branch! That meant that if this job ran on `master`, a particular plugin was getting installed, and then if it ran on `open-release/maple.master` the build would fail (since that plugin was incompatible with the source code on that branch).

Using `pip-sync` is how we avoid this issue in other situations, and for some reason we'd just never put it in place for CI.

Reply:
(Moving the `local-requirements` call to after the `testing.txt` sync is necessary so that the sync doesn't blow away the local-requirements stuff.)